Silicon/CommonSocPkg/Include/Library/GpioInfoLib.h

```diff
@@ -0,0 +1,301 @@
+/** @file
+
+  Copyright (c) 2020, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _GPIO_INFO_LIB_H_
+#define _GPIO_INFO_LIB_H_
+
+#include <Uefi/UefiBaseType.h>
+#include <GpioLibConfig.h>
+
+/**
+  This function gets Group to GPE0 configuration
+
+  @param[out] GpeDw0Value       GPIO Group to GPE_DW0 assignment
+  @param[out] GpeDw1Value       GPIO Group to GPE_DW1 assignment
+  @param[out] GpeDw2Value       GPIO Group to GPE_DW2 assignment
+**/
+VOID
+PmcGetGpioGpe (
+  OUT UINT32    *GpeDw0Value,

```

Comment:
Add EFIAPI for the library API functions. 
There are about 20 API functions in GpioInfoLib. Does the GpioLib really consume all the APIs defined in this GpioInfoLib?  Not sure if it is a burden to let the platform to provide all these APIs.

Reply:
Added EFIAPI for all interface functions.

There isnt much effort required to port these functions. All of them depend on the platform specific GPIO structures & macros. Once you have them, you can just copy paste them as globals, and use the proper names in the respective functions. ALso, there are 2 Config data related functions, which are same for all platforms. I had to put them here because I couldnt access the Gpio related config data structs from common code (they are generated inside platform folder automatically, like tag ids etc.)

Reply:
Yes, Common GpioLib is using all these functions, and they are platform specific